File: 100-Days-Series/Day 56/HollowRectangle.py

# function to print hollow rectangle pattern
def get_hollow_rectangle(rows,cols):
    for i in range(1,rows+1,1):
        for j in range(1,cols+1,1):
            if i==1 or i==rows or j==1 or j==cols:
                print('* ',end="")
            else:
                print('  ',end="")
        print()

# Let's Start here
rows=int(input('Enter the number of rows : '))
cols=int(input('Enter the number of columns : '))
get_hollow_rectangle(rows,cols)

File: 100-Days-Series/Day 56/HollowRectangle.java
public class HollowRectangle {
    // function to print hollow rectangle pattern
    public static void getHollowRectangle(int rows,int cols){
        for(int i=1;i<=rows;i++){
            for(int j=1;j<=cols;j++){
                if(i==1 || i==rows || j==1 || j==cols){
                    System.out.print("* ");
                }else{
                    System.out.print("  ");
                }
            }
            System.out.println();
        }
    }
    // Main Method
    public static void main(String [] args){
        // Let's Start here
        int rows,cols;
        java.util.Scanner s1=new java.util.Scanner(System.in);
        System.out.print("Enter the number of rows : ");
        rows=s1.nextInt();
        System.out.print("Enter the number of columns : ");
        cols=s1.nextInt();
        getHollowRectangle(rows, cols);
    }
}
